### ISSUE TITLE
Bugs when trying to play a song

### ISSUE THREAD

**Issue by Equat-ion** [vixalien/muzika#152]
When you play a song there is an uncomfortable gap and the song wont start playing until you pause and unpause it again

PS: A great app btw almost too good to be true

**Comment by gregthwuen** [comment]
There's a fix coming for that with #151 :)

**Comment by vixalien** [comment]
> When you play a song there is an uncomfortable gap

I noticed that too, And this issue is caused by a variety of factors.

When you decide to play a song, Muzika makes multiple HTTP requests to figure out the track information at once, and this is what causes the issues. Most of those requests were executed in series (i.e. one after the other) and because of the way LibSoup (the one we make network requests with) works, it would be very slow.

In the #151 PR, I plan to make all this load faster by loading everything in parallel, and better caching the results. In case you're interested, here are the requests that are done.

- `get_queue`: Gets the "queue track" metadata. This includes information such as the cover art, how to get lyrics, how to get related tracks, etc..
- `get_song`: Gets the playback URLs for the song. These are the actual URLs that will be passed to GStreamer and include separate audio and video tracks based on quality and codec (mp4 vs ogg).
- `GstDiscoverer`: When a set of URLs have been got from the previous step, we run GstDiscoverer so that GStreamer can figure out essential information from the tracks: whether the track is seekable, the duration, number of audio streams, number of video streams, etc... We could possibly do without this step, but I'm not sure. Also notice that this step **CANNOT** be made to run in parallel as it needs the result from `get_song`.
- Finally, we pass the information we got from the previous step to GStreamer, and there is a further delay as GStreamer will pre-roll and start to buffer the track before playinh.

Now GStreamer has a signal to signal that a track is almost over, and we could use this signal to load those track information quickly before the song starts playing, but unfortunately [a GJS bug causes us to not be able to receive signals from GStreamer](https://gitlab.gnome.org/GNOME/gjs/-/issues/557). Once the support for this lands, Muzika will noticeably extremely faster.
